### ISSUE TITLE
remove asserts

### ISSUE THREAD

**Issue by tarekziade** [mozilla-services/cliquet#592]
```bash
$ find cliquet -regex ".*.py" -not -regex ".*test.*" | xargs cat | grep "assert "
    assert len(errors) != 0
            assert if_none_match[0] == if_none_match[-1] == '"'
                assert if_match[0] == if_match[-1] == '"'
                assert isinstance(last_record, dict)
        assert self.match(self()), error_msg
            assert expected == current, error_msg % (expected, current)
        assert encoding == 'utf8', 'Unexpected database encoding %s' % encoding
            assert isinstance(limit, six.integer_types)  # asserted in resource
```

**Comment by almet** [comment]
Why?

**Comment by almet** [comment]
From: https://twitter.com/tarek_ziade/status/669087005582761984

**Comment by tarekziade** [comment]
Longer rational here : http://ziade.org/2015/11/25/should-i-use-pythonoptimize/

tldr: there's not compelling reason to have our app behave differently when -O or -OO is used
